h2o-algos/src/main/java/hex/anovaglm/AnovaGLM.java

```diff
@@ -0,0 +1,223 @@
+    
+    if (!(gaussian.equals(_parms._family) || tweedie.equals(_parms._family) || poisson.equals(_parms._family))) {
+      _parms._compute_p_values = false;
+      _parms._remove_collinear_columns = false;
+    }
+
+    if (nclasses() > 2)
+      error("family", " multinomial and ordinal are not supported at this point.");
+    
+    _numberOfPredCombo = calculatePredComboNumber(_numberOfPredictors, _parms._highest_interaction_term);
+    _numberOfModels = _numberOfPredCombo + 1;
+    _predNamesIndividual = extractPredNames(_dinfo, _numberOfPredictors);
+    _predictComboNames = generatePredictorCombos(_predNamesIndividual, _parms._highest_interaction_term);
+    _transformedColNames = new String[_numberOfPredCombo][];
+    _predictorColumnStart = new int[_numberOfPredCombo];
+    _degreeOfFreedom = new int[_numberOfPredCombo];
+    generatePredictorNames(_predictComboNames, _transformedColNames, _predictorColumnStart, _degreeOfFreedom, _dinfo);
+    _modelNames = generateModelNames(_predictComboNames);
+
+    if (error_count() > 0)
+      throw H2OModelBuilderIllegalArgumentException.makeFromBuilder(AnovaGLM.this);
+  }
+
+  private class AnovaGLMDriver extends Driver {
+    String[] _allTransformedColNames;   // flatten new column names
+    Key<Frame> _transformedColsKey;  // store transformed column frame key
+    Frame[] _trainingFrames;        // store generated frames
+    GLMParameters[] _glmParams;      // store GLMParameters needed to generate all the data
+    GLM[] _glmBuilder;               // store GLM Builders to be build in parallel
+    GLM[] _glmResults;
+    Frame _completeTransformedFrame;  // store transformed frame
+
+    public final void buildModel() {
+      AnovaGLMModel model = null;
+      try {
+        _dinfo = new DataInfo(_completeTransformedFrame, _valid, 1, false,
+                DataInfo.TransformType.NONE, DataInfo.TransformType.NONE,
+                _parms.missingValuesHandling() == GLMModel.GLMParameters.MissingValuesHandling.Skip,
+                _parms.imputeMissing(), _parms.makeImputer(), false, hasWeightCol(), hasOffsetCol(),
+                hasFoldCol(), null);
+        model = new AnovaGLMModel(dest(), _parms, new AnovaGLMModel.AnovaGLMModelOutput(AnovaGLM.this, _dinfo));
+        model.write_lock(_job);
+        if (_parms._save_transformed_framekeys) {
+          model._output._transformed_columns_key = _transformedColsKey.toString();
+          model._output._transformedColumnKey = _transformedColsKey;
+        }
+        _trainingFrames = buildTrainingFrames(_transformedColsKey, _numberOfModels, _transformedColNames, _parms);  // build up training frames
+        _glmParams = buildGLMParameters(_trainingFrames, _parms);
+        _job.update(1, "calling GLM to build GLM models ...");
+        _glmBuilder = buildGLMBuilders(_glmParams);
+        _glmResults = ModelBuilderHelper.trainModelsParallel(_glmBuilder, 2* H2O.NUMCPUS);
```

Comment:
what is the motivation behind using 2x #CPU?

Reply:
I copied it over from auto-ml,  they use this one.  What would you propose?

Reply:
4 models in parallel tops